src/utils/utils.js

```diff
@@ -67,15 +69,214 @@ export const getConfigObjectsForAnalytics = (configurations, groupCode) => {
+                    period,
+                    'timeliness'
+                ), // list of objects for every dataset selected (regarding completeness)
+            ],
+        },
+    }
+
+    return reportSectionsData
+}
+
+const getFacilityReportingData = (
+    reporting_rate_over_all_org_units,
+    reporting_rate_by_org_unit_level,
+    mappedConfigurations,
+    period,
+    calculatingFor
+) => {
+    // Extract key data from the analytics response
+    const headers_overall = reporting_rate_over_all_org_units.headers
+    const rows_overall = reporting_rate_over_all_org_units.rows
+    const metaData_overall = reporting_rate_over_all_org_units.metaData
+
+    const headers_level = reporting_rate_by_org_unit_level.headers
+    const rows_level = reporting_rate_by_org_unit_level.rows
+    const metaData_level = reporting_rate_by_org_unit_level.metaData
+
+    const reporting_rate_over_all_org_units_formatted =
+        getJsonObjectsFormatFromTableFormat(
+            headers_overall,
+            rows_overall,
+            metaData_overall,
+            mappedConfigurations,
+            calculatingFor
+        )
+    const reporting_rate_by_org_unit_level_formatted =
+        getJsonObjectsFormatFromTableFormat(
+            headers_level,
+            rows_level,
+            metaData_level,
+            mappedConfigurations,
+            calculatingFor
+        )
+
+    // filtering data (overall) by provided period
+    const filteredData_overall = {}
+    for (const dx in reporting_rate_over_all_org_units_formatted) {
+        if (reporting_rate_over_all_org_units_formatted[dx][period]) {
+            filteredData_overall[dx] =
+                reporting_rate_over_all_org_units_formatted[dx][period]
+        }
+    }
+
+    // filtering data (by orby provided period
+    const filteredData_levels = {}
+    for (const dx in reporting_rate_by_org_unit_level_formatted) {
+        if (reporting_rate_by_org_unit_level_formatted[dx][period]) {
+            filteredData_levels[dx] =
+                reporting_rate_by_org_unit_level_formatted[dx][period]
+        }
+    }
+
+    // updating the object with regions list, divergent count & divergent percentage
+    for (const key in filteredData_overall) {
+        const regionsWithLowScore = getRegionsWithLowScore(
+            filteredData_levels,
+            key
+        )
+        const dataset = filteredData_overall[key]
+        const dataset_levels = filteredData_levels[key] // a corresponding dataset in the reporting rates by ou level
+
+        // Calculate "divergentRegionsCount" and "divergentRegionsPercent"
+        const divergentRegionsCount = regionsWithLowScore.length
+        const totalRegionsCount = dataset_levels.length
+        const divergentRegionsPercent =
+            (divergentRegionsCount / totalRegionsCount) * 100
+
+        // Add the new properties to the dataset
+        dataset.forEach((entry) => {
+            entry.threshold = entry.threshold + '%'
```

Comment:
it might be slightly better here to leave the values as numbers and let the table decide the appropriate formatting (for example, to avoid problems with formatting in different languages)